module/localisation/BallLocalisation/src/BallLocalisation.cpp

```diff
@@ -1,40 +1,106 @@
 #include "BallLocalisation.h"
-
+#include <chrono>
 #include "extension/Configuration.h"
+#include "utility/time/time.h"
 
 #include "message/localisation/FieldObject.h"
 #include "message/vision/VisionObjects.h"
+#include "message/support/FieldDescription.h"
+#include "message/input/Sensors.h"
+
 
 #include "utility/support/eigen_armadillo.h"
+#include "utility/support/yaml_armadillo.h"
+
+#include "utility/math/matrix/Transform3D.h"
+#include "utility/math/coordinates.h"
+#include "utility/input/ServoID.h"
+#include "utility/nubugger/NUhelpers.h"
 
 namespace module {
 namespace localisation {
 
     using extension::Configuration;
-
+    using utility::time::TimeDifferenceSeconds;
     using message::localisation::Ball;
+    using message::support::FieldDescription;
+    using message::input::Sensors;
+
+    using utility::math::matrix::Transform3D;
+    using utility::math::coordinates::cartesianToSpherical;
+    using ServoID = utility::input::ServoID;
+    using utility::nubugger::graph;
 
     BallLocalisation::BallLocalisation(std::unique_ptr<NUClear::Environment> environment)
-        : Reactor(std::move(environment)), filter() {
+    : Reactor(std::move(environment))
+    , filter() {
+
+        last_measurement_update_time = NUClear::clock::now();
+        last_time_update_time = NUClear::clock::now();
 
-        on<Configuration>("BallLocalisation.yaml").then([this](const Configuration&) {
+        on<Configuration>("BallLocalisation.yaml").then([this] (const Configuration& config) {
             auto message = std::make_unique<std::vector<Ball>>();
-            emit(message);
+        	emit(message);
             emit(std::make_unique<Ball>());
 
+            filter.model.processNoiseDiagonal = config["process_noise_diagonal"].as<arma::vec>();
+
             // Use configuration here from file BallLocalisation.yaml
         });
 
-        on<Trigger<std::vector<message::vision::Ball>>>().then([this](const std::vector<message::vision::Ball>& balls) {
-            if (balls.size() > 0) {
-                auto message = std::make_unique<std::vector<Ball>>();
-                message->push_back(Ball());
-                message->back().locObject.last_measurement_time = NUClear::clock::now();
-                message->back().locObject.position              = balls[0].torsoSpacePosition.head<2>();
-                emit(std::make_unique<Ball>(message->back()));
-                emit(message);
-            }
+        /* Run Time Update */
+        on<Every<15, Per<std::chrono::seconds>>, Sync<BallLocalisation>
+         , With<FieldDescription>
+         , With<Sensors>>().then("BallLocalisation Time", [this](
```

Comment:
You shouldn't be using these sensors!, Use the ones that are attached to the ball object as they were taken when the image was.